### PR TITLE
Pbi 193 graphic vars from csv

### PR DESCRIPTION
**Pull request by GagsdName**
Added functionality to load data variable to graphic variable mappings from csv fields

### CODE REVIEW

File: projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts [hunk 1]
```diff
@@ -98,36 +106,72 @@ export class CSVTemplateProject extends DefaultProject {
...
+  getPredefinedMappings(mappingFields: string[]): {} {
+    const predefinedMappings = {};
+
+    for (const field of mappingFields) {
+      const splitFields = field.trim().split('$$');
+      if (splitFields.length > 1) {
+        if (!predefinedMappings[splitFields[0]]) {
+          predefinedMappings[splitFields[0]] = {};
+        }
+        predefinedMappings[splitFields[0]][splitFields[1]] = [{selector: splitFields[0]}];
```
Comment (bherr2) [inline projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts]: Have you tested this? It looks wrong. I believe selector: should be `selector: field`.

Comment (GagsdName) [reply projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts]: Yes. The 'field' would be of type 'dv$$gv' and splitField[0] would be 'dv'.

File: projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import { GraphicSymbol } from '../../shared/graphic-symbol';
 import { Visualization } from '../../shared/visualization';
 import { ScatterplotVisualization } from '../ngx-dino/visualizations';
 import { DefaultGraphicSymbol } from './default-graphic-symbol';
+import { filter } from 'rxjs/operators';
```
Comment (bherr2) [inline projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts]: unused import

File: projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts [hunk 3]
```diff
@@ -88,7 +96,7 @@ export class CSVTemplateProject extends DefaultProject {
         labelPlural: 'CSV Data',
         description: fileName || undefined,
         defaultRecordStream: 'csvData',
-        dataVariables: this.fields.map(f => {
+        dataVariables: this.getFilteredFields(this.fields).map(f => {
```
Comment (bherr2) [inline projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts]: This doesn't add data variables for columns with "xxx$$yyy" where xxx is not in the list of fields. Ie if they mispelled xxx or they just want to have certain gv's and nothing else for a dv. In this case a new field xxx should be added. Setting up GVs should still stay the same.

File: projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts [hunk 4]
```diff
@@ -46,6 +47,13 @@ export class CSVTemplateProject extends DefaultProject {
       new DefaultRawData({id: 'csvRawData', template: 'json', data: {csvData: parseResults.data}})
     ];
   }
+
+  private getFilteredFields(fields: string[]): string[] {
```
Comment (bherr2) [inline projects/dvl-fw/src/lib/plugins/default/csv-template-project.ts]: Don't like the name. getDataVariableNames?

**Pull request merged**